libs/common/src/types/mongo-db-models/tfm-deals.ts

```diff
@@ -0,0 +1,17 @@
+import { ObjectId } from 'mongodb';
+import { TfmDealCancellation } from '../tfm-deal-cancellation';
+import { Deal } from './deals';
+import { AuditDatabaseRecord } from '../audit-database-record';
+
+/**
+ * Type of the mongo db "tfm-deals" collection
+ *
+ * This type is likely incomplete and should be added
+ * to as and when new properties are discovered
+ */
+export type TfmDeal = {
+  _id: ObjectId;
+  dealSnapshot: Deal;
+  tfm: { cancellation: TfmDealCancellation };
+  auditRecord: AuditDatabaseRecord;
```

Comment:
@MarRobSoftwire can I get your input here? 👀

Reply:
This is a good point, I suppose it is optional (but will exist on any deal made after audit logs is released). It's nice that this will cause a type error if you try to create a deal without `auditRecord`. The code doesn't actually use the auditRecord on any `findOne` methods or similar - its just sent to the data lake.

If we do want to make it optional it would be good to have a comment explaining that its only optional on in flight deals, pre audit log release

```suggestion
  // Audit records may not exist on a deal if it has not been modified after Audit Logs is released
  auditRecord?: AuditDatabaseRecord;
```